### DISCUSSION TITLE
Thanks!

### DISCUSSION THREAD

**Discussion by dieselvero** [codecrafting-io/AdvancedPathTracingCP2077#1] [General]
Didn't know where else to put this, but thanks for making this and glad I found it...more or less by accident.  I like having these options and enhancement for path tracing but without the heavy handed approach of the Ultra+ mod, which changes a lot of things and often ends up with issues as a result.  Great job!

**Comment by codecrafting-io** [comment]
Thank you very much. Just out of curiosity, how are you using this mod? I am currently working on a revision of the quality settings to reduce the boiling/crawling effect on some settings. About game patch 2.13, CET just released a supported version, but I haven't tested it yet, but hope it still works.

**Reply by dieselvero** [reply]
I use the Ultra setting with ReSTIR DI/ ReGIR GI because I find that it gives the best picture in terms of sharpness.  I'd like to use the ReGIR for both, but the picture just gets too muddy with RR on.  I don't use psycho quality because it kills my frames, even with a 4090.

One thing I did notice, however, was that when using any setting aside from "Vanilla" will cause the brightness to go up quite considerably.  Meaning, it seems like there is more illumination.  I mean, for quality setting, anything aside from "Vanilla" and for Mode, anything other than "ReSTIR DI/GI".   See the screens below.  First one is vanilla setting and second is Ultra.

![Vanilla](https://github.com/user-attachments/assets/e3e2e18d-66d0-472a-8353-ade5ca2d1916)

![Ultra](https://github.com/user-attachments/assets/8091d198-7643-4c38-bc1a-69461ee483f4)

**Reply by dieselvero** [reply]
I would think that increasing the number of bounces might do this since the light would reflect more, but the bounce number is the same in both shots.

**Reply by codecrafting-io** [reply]
The number of bounces only affects the ReSTIR DI mode. The brighter image is probably related to the optimizations internal setting "Editor/RTXDI/EmissiveDistanceThreshold" which goes from 0.5 to 0.67. I find this setting helps picture quality in some situations, but if you don't like it you can either disable the optimizations or edit the file `ptQuality.lua` in Optimizations > [true] session. You can play around with the CET console using the command `GameOptions.Set('Editor/RTXDI', 'EmissiveDistanceThreshold', '0.67')`.

I also find problems with specular light using ReGIR DI/GI, you loose detail on those surfaces for some reason, so just recently I created this ReSTIR DI + ReGIR GI mode, which is heavier than full ReGIR, I guess because ReGIR is one algorithm and using both is more expensive. Also, with full ReGIR you have noise break ups where some surfaces oscilate the light so much that RR kind breaks up and shows terrible noise, which doesn't happen with ReSTIR DI + ReGIR GI.

Psycho Quality Settings It's kind of a meme setting or for photo mode only, so you "have to be a psycho" to enable it 😅.

I just tested with 2.13 and everything is ok, you just need to update to the latest CET version

**Reply by dieselvero** [reply]
I'm so sorry, it wasn't your mod. I have figured out what it is.  I am using Nova City and the ini file has some changes to Raytracing in it.  When I commented those out, the brighter illumination exists even on Vanilla.  

The settings were EnableRIS=false and EnableImportanceSampling=false.

**Reply by paperclip-dayo** [reply]
Hey, just a quick question, what settings are also shared with the normal Raytracing? 
I've noticed that "optimizations" have an affect: https://imgsli.com/MzM1NjE3 (bottom right shacks/AC and the tower with Kiroshi sign). At quick glance nothing else seems to change, well aside from the self-reflection option.

**Comment by codecrafting-io** [comment]
Hi, these settings will affect normal RayTracing, you can see on the ptSettings.lua file:

```
 {
            {
                category    = "Editor/Denoising/NRD",
                name        = "DisocclusionThreshold",
                value       = "0.002"
            },
            {
                category    = "RayTracing",
                name        = "EnableImportanceSampling",
                value       = "false"
            },
            {
                category    = "RayTracing",
                name        = "TransparentReflectionEnvironmentBlendFactor",
                value       = "0.8"
            },
            {
                category    = "RayTracing/Reflection",
                name        = "EnableHalfResolutionTracing",
                value       = "0"
            }
        }
    }
}
```

The main difference that you are seeing is probably related to DisocclusionThreshold
